scripts/limonade/src/index.ts

```diff
@@ -197,6 +267,71 @@ async function run() {
     }, argv.solanaWebsocketTimeout);
   });
 
+  if (priceConfigs.length > 0) {
+    const hermesClient = new HermesClient(argv.hermesEndpoint, {});
+
+    const eventSource = await hermesClient.getPriceUpdatesStream(
+      priceConfigs.map((priceConfig) => priceConfig.pythFeedId),
+      {
+        encoding: "hex",
+        parsed: true,
+        ignoreInvalidPriceIds: true,
+      }
+    );
+
+    eventSource.onerror = (event: Event) => {
+      console.error("Hermes streaming error", event);
+    };
+
+    /// Await for the first message before continuing
+    await new Promise<void>((resolve, reject) => {
+      setTimeout(() => {
+        reject(new Error("Hermes streaming timeout"));
+      }, argv.hermesStreamingTimeout);
```

Comment:
there should be a way to write this with a single `setTimeout`, since the error message is not distinct between the two cases (and the timeout durations are the same). any reason you wanted to distinguish between these two?

Reply:
I don't see a way to write this with a single timeout.
The first one is handling the case where we never receive any messages.
The second one handles if we've already received messages in the past but we stop receiving them.